tests/components/test_streams_app.py

```diff
@@ -178,6 +178,27 @@ def test_should_validate(self, config: PipelineConfig, handlers: ComponentHandle
                 },
             )
 
+        with pytest.raises(ValueError):
```

Comment:
add a comment why this fails. I am assuming because role is not allowed for type input

Reply:
You are right in your assumption, I will.